api/envoy/config/filter/http/router/v2/BUILD

```diff
@@ -1,4 +1,4 @@
-load("@envoy_api//bazel:api_build_system.bzl", "api_proto_library_internal")
+load("@envoy_api//bazel:api_build_system.bzl", "api_go_proto_library", "api_proto_library_internal")
```

Comment:
Added this because we want to validate values set for `strict_check_headers` in the config in  [api/envoy/config/filter/http/router/v2/router.proto#L14](https://github.com/envoyproxy/envoy/blob/ed2a492bd1012ce35fc3af9df6123747eb31cc3b/api/envoy/config/filter/http/router/v2/router.proto#L14), which requires inclusion of `com_envoyproxy_protoc_gen_validate/validate/validate.proto`.

Reply:
I'm no bazel expert but I thought this was only necessary if you directly referenced the `api_go_proto_library` rule in this BUILD file which it doesn't seem like you're doing? I could be totally wrong here, so feel free to leave it if it's necessary for the build

Reply:
Will repush with this removed and add it back if it turns out to actually result in failed builds.

Hmm.. I rebuilt `//source/exe:envoy-static` (not from a clean build though) with this change removed and things seemed to work. There are other places in `api/envoy/config/filter/http/**` that reference proto validation without this declaration in their `BUILD` files.